src/v/cluster/scheduling/leader_balancer.cc

```diff
@@ -442,9 +446,55 @@ bool leader_balancer::should_stop_balance() const {
 }
 
 bool leader_balancer::leadership_pinning_enabled() const {
+    if (_feature_table.should_sanction()) {
+        const auto warning_msg = leadership_pinning_sanction_warning_msg();
+        if (warning_msg.has_value()) {
+            vlog(clusterlog.warn, "{}", warning_msg.value());
+        }
+        return false;
+    }
+
     return _enable_rack_awareness();
 }
 
+std::optional<ss::sstring>
+leader_balancer::leadership_pinning_sanction_warning_msg() const {
+    const auto& [_, config_prop_sanctioned] = _default_preference(true);
+    chunked_vector<model::topic_namespace> sanctioned_topics;
+
+    for (const auto& topic : _topics.topics_map()) {
+        const auto& preference
+          = topic.second.get_configuration().properties.leaders_preference;
+        if (
+          preference.has_value()
+          && config::shard_local_cfg()
+               .default_leaders_preference.check_restricted(
+                 preference.value())) {
+            sanctioned_topics.push_back(topic.first);
+        }
+    }
```

Comment:
There is a logic overlap between this function and the `feature_table::leadership_pinning_enabled`. It is worth addressing in a follow up refactoring, but i think this should be enough for this iteration.